src/seraphis_mocks/scan_context_async_mock.cpp

```diff
@@ -0,0 +1,797 @@
+    // TODO: pre-RCT outputs yield incorrect values here but this is only used for spending
+    // need https://github.com/UkoeHB/monero/pull/40 in order to handle pre-RCT outputs
+    return !output_indices.empty() ? output_indices[0] : 0;
+}
+//-------------------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------------------
+static void prepare_unscanned_legacy_transaction(const crypto::hash &tx_hash,
+    const cryptonote::transaction &tx,
+    const uint64_t total_output_count_before_tx,
+    sp::LegacyUnscannedTransaction &unscanned_tx_out)
+{
+    unscanned_tx_out = LegacyUnscannedTransaction{};
+
+    unscanned_tx_out.transaction_id = rct::hash2rct(tx_hash);
+    unscanned_tx_out.total_enotes_before_tx = total_output_count_before_tx;
+    unscanned_tx_out.unlock_time = tx.unlock_time;
+
+    unscanned_tx_out.tx_memo = sp::TxExtra(
+            (const unsigned char *) tx.extra.data(),
+            (const unsigned char *) tx.extra.data() + tx.extra.size()
+        );
+
+    sp::legacy_outputs_to_enotes(tx, unscanned_tx_out.enotes);
+
+    unscanned_tx_out.legacy_key_images.reserve(tx.vin.size());
+    for (const auto &in: tx.vin)
+    {
+        if (in.type() != typeid(cryptonote::txin_to_key))
+            continue;
+        const auto &txin = boost::get<cryptonote::txin_to_key>(in);
+        unscanned_tx_out.legacy_key_images.emplace_back(txin.k_image);
+    }
+}
+//-------------------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------------------
+static bool is_terminal_chunk(const sp::scanning::ChunkContext &context, const std::uint64_t end_scan_index)
+{
+    if (sp::scanning::chunk_context_is_empty(context))
+    {
+        MDEBUG("Chunk context is empty starting at " << context.start_index);
+        return true;
+    }
+
+    // is the chunk the terminal chunk in the chain
+    const std::uint64_t current_chunk_end_index{context.start_index + sp::scanning::chunk_size(context)};
+    if (current_chunk_end_index >= end_scan_index)
+    {
+        MDEBUG("Chunk context end index: " << current_chunk_end_index
+            << " (end_scan_index=" << end_scan_index << ")");
+        return true;
+    }
+
+    return false;
+}
+//-------------------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------------------
+static void rpc_get_blocks_internal(const ChunkRequest &chunk_request,
+    const std::function<bool(const cryptonote::COMMAND_RPC_GET_BLOCKS_FAST::request&, cryptonote::COMMAND_RPC_GET_BLOCKS_FAST::response&)> &rpc_get_blocks,
```

Comment:
```suggestion
    const std::function<bool(
        const cryptonote::COMMAND_RPC_GET_BLOCKS_FAST::request&,
        cryptonote::COMMAND_RPC_GET_BLOCKS_FAST::response&
    )> &rpc_get_blocks,
```
Just noticed it now, but this callback is way outside max width.